src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -46,6 +46,12 @@ public sealed class AgentSettings
 
         [DataMember(EmitDefaultValue = false)]
         public int DeploymentGroupId { get; set; }
+
+        [DataMember(EmitDefaultValue = false)]
+        public string ProjectId { get; set; }
```

Comment:
@chshrikh does this means both setting will needs be there forever? that is something i am not really want to see.
basically you change the projectname to be projectid in your REST contract, but you don't provide in client to migrate to use the new contract, instead you leave the client in hyper mode.